From ab0ee597c569b9d5a371d14a65736fa74913658a Mon Sep 17 00:00:00 2001
From: Arun Bharadwaj <arun@gumstix.com>
Date: Thu, 19 May 2016 13:24:00 -0700
Subject: [PATCH 47/57] am335x-pepper-43c: Updated device tree for v12

This patch contains the following changes:
  * Captouch has moved to i2c2
  * Audio codec has moved to i2c1
  * IMU module has been changed
  * Enable red LED as a "heardbeat" trigger
  * Enable blue LED as an SD card triggger

Signed-off-by: Arun Bharadwaj <arun@gumstix.com>
---
 arch/arm/boot/dts/am335x-pepper-43c.dts | 92 ++++++++++++++++++---------------
 1 file changed, 49 insertions(+), 43 deletions(-)

diff --git a/arch/arm/boot/dts/am335x-pepper-43c.dts b/arch/arm/boot/dts/am335x-pepper-43c.dts
index 45b782e..cb398b4 100644
--- a/arch/arm/boot/dts/am335x-pepper-43c.dts
+++ b/arch/arm/boot/dts/am335x-pepper-43c.dts
@@ -101,12 +101,6 @@
 		reg = <0x50>;
 	};
 
-	audio_codec: tlv320aic3106@1b {
-		compatible = "ti,tlv320aic3106";
-		reg = <0x1b>;
-		ai3x-micbias-vg = <0x2>;
-	};
-
 	ds_rtc: ds1340@68 {
 		compatible = "dallas,ds1340";
 		reg = <0x68>;
@@ -117,43 +111,52 @@
 &i2c1 {
 	status = "okay";
 	pinctrl-names = "default";
-	pinctrl-0 = <&i2c1_pins>;
+	pinctrl-0 = <&i2c1_pins &imu_pins>;
 	clock-frequency = <400000>;
+	gpios = <&gpio1 2 GPIO_ACTIVE_HIGH>;
 
+	audio_codec: tlv320aic3106@1b {
+		compatible = "ti,tlv320aic3106";
+		reg = <0x1b>;
+		ai3x-micbias-vg = <0x2>;
+	};
 
-	// no device tree bindings or driver for this chip...fake it
-	accel: mpu9150@69 {
-		compatible = "invensense,mpu6050";
-		pinctrl-names = "default";
-		pinctrl-0 = <&accel_pins>;
-		reg = <0x69>;
-		interrupt-parent = <&gpio1>;
-		interrupt = <2 1>;
+	/* Accelerometer/Magnetometer */
+	lsm303d: lsm303d@1e {
+		compatible = "lsm303d";
+		int1-gpio = <&gpio2 4 GPIO_ACTIVE_HIGH>;
+		int2-gpio = <&gpio0 30 GPIO_ACTIVE_HIGH>;
+		reg = <0x1e>;
 	};
 
-        captouch: edt-ft5306@38 {
-                status = "okay";
-                compatible = "edt,edt-ft5306", "edt,edt-ft5x06";
-                pinctrl-names = "default";
-                pinctrl-0 = <&captouch_pins>;
+	/* Gyroscope */
+	l3gd20h: l3gd20h@6a {
+		compatible = "l3gd20h_gyr";
+		int-gpio = <&gpio1 2 GPIO_ACTIVE_HIGH>;
+		reg = <0x6a>;
+	};
+};
 
-                reg = <0x38>;
-                interrupt-parent = <&gpio0>;
-                interrupts = <20 0>;
+&i2c2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c2_pins>;
+	clock-frequency = <400000>;
 
-                wake-gpios = <&gpio0 5 GPIO_ACTIVE_HIGH>;
+	captouch: edt-ft5306@38 {
+		status = "okay";
+		compatible = "edt,edt-ft5306", "edt,edt-ft5x06";
+		pinctrl-names = "default";
+		pinctrl-0 = <&captouch_pins>;
 
-                touchscreen-size-x = <480>;
-                touchscreen-size-y = <272>;
-        };
-};
+		reg = <0x38>;
+		interrupt-parent = <&gpio0>;
+		interrupts = <20 0>;
 
-/* Accelerometer */
-&am33xx_pinmux {
-	accel_pins: pinmux_accel {
-		pinctrl-single,pins = <
-			0x08 (PIN_INPUT | MUX_MODE7)   /* gpmc_ad2.gpio1_2 */
-		>;
+		wake-gpios = <&gpio0 5 GPIO_ACTIVE_HIGH>;
+
+		touchscreen-size-x = <480>;
+		touchscreen-size-y = <272>;
 	};
 };
 
@@ -632,13 +635,13 @@
 	led@1 {
 		label = "pepper:user0:red";
 		gpios = <&gpio1 21 GPIO_ACTIVE_HIGH>;
-		linux,default-trigger = "none";
+		linux,default-trigger = "heartbeat";
 		default-state = "off";
 	};
 	led@0 {
 		label = "pepper:user0:blue";
 		gpios = <&gpio1 20 GPIO_ACTIVE_HIGH>;
-		linux,default-trigger = "none";
+		linux,default-trigger = "mmc0";
 		default-state = "off";
 	};
 };
@@ -681,14 +684,17 @@
 			0x16C (PIN_INPUT_PULLUP | MUX_MODE3)	/* uart0_rtsn.i2c1_scl */
 		>;
 	};
-	sd_pins: pinmux_sd_card {
+	i2c2_pins: pinmux_i2c2 {
 		pinctrl-single,pins = <
-			0xf0 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat0.mmc0_dat0 */
-			0xf4 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat1.mmc0_dat1 */
-			0xf8 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat2.mmc0_dat2 */
-			0xfc (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat3.mmc0_dat3 */
-			0x100 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_clk.mmc0_clk */
-			0x104 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_cmd.mmc0_cmd */
+			0x150 (PIN_INPUT_PULLUP | MUX_MODE2)	/* spi0_sclk.i2c2_sda */
+			0x154 (PIN_INPUT_PULLUP | MUX_MODE2)	/* spi0_d0.i2c2_scl */
+		>;
+	};
+	imu_pins: pinmux_imu_pins {
+		pinctrl-single,pins = <
+			0x8  (PIN_INPUT | MUX_MODE7)	/* gpmc_ad2.gpio1_2 */
+			0x98 (PIN_INPUT | MUX_MODE7)	/* gpmc_wen.gpio2_4 */
+			0x70 (PIN_INPUT | MUX_MODE7)	/* gpmc_wait0.gpio0_30 */
 		>;
 	};
 };
-- 
2.7.4

